text/chapter3.md

```diff
@@ -639,11 +639,16 @@ and the right associativity of our new `++` operator lets us get rid of the pare
 book6 = john ++ peggy ++ ned ++ emptyBook
 ```
 
-Likewise, in the code for `findEntry` above, we used a different form of function application: the `head` function was applied to the expression `filter filterEntry book` by using the infix `$` symbol.
+Another common technique for eliminating parens is to use `apply`'s infix operator `$`, along with your standard prefix functions.
+
+For example, the earlier `book3` example could be rewritten as:
+```haskell
+book7 = insertEntry john $ insertEntry peggy $ insertEntry ned emptyBook
```

Comment:
Relocated this example closer to the previous snippet so we can avoid the recap.

Also a fan of showing the most compelling example first, then the deeper explanation. So I did some other reordering. I think many beginning users can just make a mental note of what `$` means in practice, and be productive with it without necessarily being able to write the code for `apply` if quizzed.

Reply:
Yes, this is much tidier.